### PR TITLE
BCType::foextrap as default for (density, BC::slip_wall) combination when advection_type == "BDS"

### PR DESCRIPTION
**Pull request by siddanib**
Making `BCType::foextrap` as the default type for _density_ and _tracer_ at `BC::slip_wall` boundary only when `advection_type == "BDS"`. Also made a minor change in docs (`src/boundary_conditions/README.org`) for _(scalar/density, sw)_ combination.

### CODE REVIEW

**Review by ajnonaka — APPROVED** [review]

**Comment by ajnonaka** [comment]
Hold off on merging - we should consider making all scalars operate the same way.

File: src/boundary_conditions/README.org
Comment (ajnonaka) [inline src/boundary_conditions/README.org]:
```suggestion
| rho    | foextrap | foextrap | ext_dir  | foextrap    | if (advection_type == "BDS") foextrap |
|        |          |          |          |             | else hoextrap                         |
| scalar    | foextrap | foextrap | ext_dir  | foextrap    | if (advection_type == "BDS") foextrap |
|        |          |          |          |             | else hoextrap                         |
```

File: src/boundary_conditions/boundary_conditions.cpp
Comment (ajnonaka) [inline src/boundary_conditions/boundary_conditions.cpp:274]: Can you apply this to scalars and tracers as well, and leave a comment "// BDS requires foextrap to avoid introduction of local max/min"

**Review by ajnonaka — APPROVED** [review]
Looks good - thanks Bhargav!  Let me try running the CI suite before merging.

**Comment by siddanib** [comment]
Thank you for the suggestions, Andy (@ajnonaka)! I believe I might have to rebase based on #114 to pass all the checks.

**Comment by cgilet** [comment]
The velocity uses hoextrap for the tangential components for slip wall. Would that also be incorrect with BDS?

**Comment by ajnonaka** [comment]
hoextrap for BDS + velocity is fine - we don't have local min/max constraints.

**Pull request merged**
